rice/Arg.ipp

```diff
@@ -59,69 +55,37 @@ namespace Rice {
       template<typename Arg_Type>
       Arg& operator=(Arg_Type val)
       {
-        defaultValue = new type<Arg_Type>(val);
+        this->defaultValue_ = val;
         return *this;
       }
 
       //! Check if this Arg has a default value associated with it
-      bool hasDefaultValue() const {
-        return defaultValue != 0;
+      bool hasDefaultValue() const
+      {
+        return this->defaultValue_.has_value();
       }
 
-      //! Return the default value associated with this Arg
+      //! Return a reference to the default value associated with this Arg
       /*! \return the type saved to this Arg
        */
       template<typename Arg_Type>
-      Arg_Type getDefaultValue()
+      Arg_Type& defaultValue()
       {
-        return static_cast< type<Arg_Type>* >(defaultValue)->held;
+        return std::any_cast<Arg_Type&>(this->defaultValue_);
       }
 
       //! Get the name of this Arg
-      const char* name() const
+      const std::string name() const
       {
         return name_;
       }
 
     private:
-
       //! Name of the argument
-      const char* name_;
-
-      /**
-       * The following is a stripped down version of
-       * Boost.Any.
-       */
-
-      class type_base
-      {
-        public:
-          virtual ~type_base() {}
-          virtual type_base* clone() const = 0;
-      };
-
-      template<typename Type>
-      class type : public type_base
-      {
-        public:
-          type(Type value)
-            :held(value)
-          {}
-
-          virtual ~type() { }
-
-          virtual type_base* clone() const
-          {
-            return new type(held);
-          }
-
-          Type held;
-      };
-
-    public:
+      const std::string name_;
 
       //! Our saved default value
-      type_base* defaultValue;
+      std::any defaultValue_;
```

Comment:
Love the C++ has this type built in now!

Reply:
Yes, very handy!